tests/std/tests/GH_000311_enforce_core_headers/test.compile.pass.cpp

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CORE_HEADERS_ONLY
+#define _ENFORCE_ONLY_CORE_HEADERS

```

Comment:
I think we could simply fuse this into `GH_001411_core_headers`. That test also uses the `usual_matrix` and defines `_CORE_HEADERS_ONLY`, so it'll provide the same coverage.

Rationale: Every added test directory incurs a minor cost. It's okay to add new test directories for new features and bugfixes - we definitely went overboard in the past by cramming tons of different tests into a single file (I can't remember what I was thinking with `Dev11_0000000_user_defined_literals` :joy_cat:) - but when we can slightly extend an existing test with no loss of coverage, it's simpler and faster to do that.